service/integration/resource_mappings_test.go

```diff
@@ -112,6 +389,110 @@ func (s *ResourceMappingsSuite) Test_ListResourceMappings() {
 	}
 }
 
+func (s *ResourceMappingsSuite) Test_ListResourceMappingsByGroupId() {
+	req := &resourcemapping.ListResourceMappingsRequest{
+		GroupId: s.getResourceMappingGroupFixtures()[0].ID,
+	}
+	mappings, err := s.db.PolicyClient.ListResourceMappings(s.ctx, req)
+	s.Require().NoError(err)
+	s.NotNil(mappings)
+	for _, mapping := range mappings {
+		expectedGroupID := req.GetGroupId()
+		actualGroupID := mapping.GetGroup().GetId()
+		s.Equal(expectedGroupID, actualGroupID,
+			fmt.Sprintf("expected group id %s, got %s", expectedGroupID, actualGroupID))
+	}
+}
+
+func (s *ResourceMappingsSuite) Test_ListResourceMappingsByGroupFqns() {
+	scenarioDotComNs := s.getScenarioDotComNamespace()
+	scenarioDotComGroup := s.f.GetResourceMappingGroupKey("scenario.com_ns_group_1")
+	scenarioDotComGroupMapping := s.f.GetResourceMappingKey("resource_mapping_to_attribute_value3")
+
+	groupFqn := fmt.Sprintf("https://%s/resm/%s", scenarioDotComNs.Name, scenarioDotComGroup.Name)
+
+	fqnRmGroupMap, err := s.db.PolicyClient.ListResourceMappingsByGroupFqns(s.ctx, []string{groupFqn})
+	s.Require().NoError(err)
+	s.NotNil(fqnRmGroupMap)
+
+	mappingsByGroup, ok := fqnRmGroupMap[groupFqn]
+	s.True(ok)
+	s.NotNil(mappingsByGroup)
+	s.Equal(scenarioDotComGroup.ID, mappingsByGroup.GetGroup().GetId())
+	s.Equal(scenarioDotComGroup.NamespaceID, mappingsByGroup.GetGroup().GetNamespaceId())
+	s.Equal(scenarioDotComGroup.Name, mappingsByGroup.GetGroup().GetName())
+
+	s.Equal(len(mappingsByGroup.Mappings), 1)
```

Comment:
The reason for the linter complaining about direct access to fields is that they can be unexpectedly `nil` and lead to panics. Using the proto accessors such as `GetMappings()` instead of `.Mappings` will always avoid a potential nil pointer panic and allow nesting without error such as `parent.GetChild().GetParent().GetChild()`

Reply:
Ah, good catch.  That was a typo and I meant to use `GetMappings`.